docs/img/src/architecture.puml

```diff
@@ -0,0 +1,39 @@
+@startuml
+skinparam componentStyle rectangle
+
+package "OpenTelementry connector" {
+  [Processing data]
+  package "DataSetExporter" {
+    [Converts OTel structure into events] as convTraces
+    [Converts OTel structure into events] as convLogs
+    package "DataSet Go library" {
+      [Publishes events into topic based on group_by] as Publisher
+      [Pub/Sub for events - topic based on group_by] as PubSubEvents
+      [consumes events, stores them in buffers, publises buffer in topic based on group_by] as EventsConsumer
+      [Pub/Sub for buffers - topic based on group_by] as PubSubBuffers
+      [consumes buffers, calls DataSet API, retries in case of error] as BufferConsumer
+      [Buffer sweeper - publishes old buffers] as BufferSweeper
+    }
+  }
+}
+
+package "DataSet API servers" {
+  [/AddEvents REST endpoint] as DataSetAddEvents
```

Comment:
Minor thing - actual API endpoint path is either ``/addEvents`` or ``/api/addEvents`` (lower case `a`, the later is preferred path).